pydoctor/templatewriter/writer.py

```diff
@@ -31,67 +38,86 @@ def __subclasshook__(cls, subclass: Type[object]) -> bool:
         os.makedirs(self.base, exist_ok=True)
-        shutil.copyfile(templatefile('apidocs.css'),
-                        os.path.join(self.base, 'apidocs.css'))
-        shutil.copyfile(templatefile('bootstrap.min.css'),
-                        os.path.join(self.base, 'bootstrap.min.css'))
-        shutil.copyfile(templatefile('pydoctor.js'),
-                        os.path.join(self.base, 'pydoctor.js'))
+        shutil.copy(
+            self.template_lookup.get_template('apidocs.css').path,
+            self.base.joinpath('apidocs.css'))
+        shutil.copy(
+            self.template_lookup.get_template('extra.css').path,
+            self.base.joinpath('extra.css'))
+        shutil.copy(
+            self.template_lookup.get_template('bootstrap.min.css').path,
+            self.base.joinpath('bootstrap.min.css'))
+        shutil.copy(
+            self.template_lookup.get_template('pydoctor.js').path,
+            self.base.joinpath('pydoctor.js'))
 
-    def writeIndividualFiles(self, obs):
+    def writeIndividualFiles(self, obs:Iterable[model.Documentable]) -> None:
+        """
+        Iterate trought C{obs} and call L{_writeDocsFor} method for each L{Documentable}. 
+        """
         self.dry_run = True
         for ob in obs:
             self._writeDocsFor(ob)
         self.dry_run = False
         for ob in obs:
             self._writeDocsFor(ob)
 
-    def writeModuleIndex(self, system):
+    def writeModuleIndex(self, system:model.System) -> None:
         import time
-        for i, pclass in enumerate(summary.summarypages):
+        for pclass in summary.summarypages:
             system.msg('html', 'starting ' + pclass.__name__ + ' ...', nonl=True)
             T = time.time()
-            page = pclass(system)
-            f = open(os.path.join(self.base, pclass.filename), 'wb')
-            flattenToFile(f, page)
-            f.close()
+            page = pclass(system=system, template_lookup=self.template_lookup)
+            with self.base.joinpath(pclass.filename).open('wb') as fobj:  
+                flattenToFile(fobj, page)
             system.msg('html', "took %fs"%(time.time() - T), wantsnl=False)
 
-    def _writeDocsFor(self, ob):
+    def _writeDocsFor(self, ob:model.Documentable) -> None:
         if not ob.isVisible:
             return
         if ob.documentation_location is model.DocLocation.OWN_PAGE:
             if self.dry_run:
                 self.total_pages += 1
             else:
-                path = FilePath(self.base).child(f'{ob.fullName()}.html')
-                with path.open('wb') as out:
-                    self._writeDocsForOne(ob, out)
+                with self.base.joinpath(f'{ob.fullName()}.html').open('wb') as fobj:
+                    self._writeDocsForOne(ob, fobj)
         for o in ob.contents.values():
             self._writeDocsFor(o)
 
-    def _writeDocsForOne(self, ob, fobj):
+    def _writeDocsForOne(self, ob:model.Documentable, fobj:IO[bytes]) -> None:
```

Comment:
Thanks for adding all the new type annotations.

Reply:
You're welcome :)